src/us/kbase/test/auth2/MongoController.java

```diff
@@ -0,0 +1,147 @@
+            new LinkedList<String>();
+    static {
+        tempDirectories.add(DATA_DIR);
+    }
+
+    private final static Version MONGO_DB_3 =
+            Version.forIntegers(3,6,23);
+
+    private final Path tempDir;
+
+    private final Process mongo;
+
+    private final int port;
+
+    public MongoController(
+            final String mongoExe,
+            final Path rootTempDir)
+            throws Exception {
+        this(mongoExe, rootTempDir, false);
+    }
+
+    public MongoController(
+            final String mongoExe,
+            final Path rootTempDir,
+            final boolean useWiredTiger)
+            throws Exception {
+        checkExe(mongoExe, "mongod server");
+        tempDir = makeTempDirs(rootTempDir, "MongoController-", tempDirectories);
+        port = findFreePort();
+        Version dbVer = getMongoDBVer(mongoExe);
+        List<String> command = getMongoServerStartCommand(mongoExe, useWiredTiger, dbVer);
+        mongo = startProcess(command);
+    }
+
+    public int getServerPort() {
+        return port;
+    }
+
+    public Path getTempDir() {
+        return tempDir;
+    }
+
+    public void destroy(boolean deleteTempFiles) throws IOException {
+        if (mongo != null) {
+            mongo.destroy();
+        }
+        if (tempDir != null && deleteTempFiles) {
+            FileUtils.deleteDirectory(tempDir.toFile());
+        }
+    }
+
+    public static Version getMongoDBVer(final String mongoExe) throws IOException {
+
+        // build MongoDB version check command
+        List<String> command = new LinkedList<String>();
+        command.addAll(Arrays.asList(mongoExe, "--version"));
+
+        // start MongoDB version check process
+        ProcessBuilder checkVerPb = new ProcessBuilder(command);
+        Process checkVerProcess  = checkVerPb.start();
+
+        // parse mongod --version output string
+        String dbVer = new BufferedReader(
+                new InputStreamReader(checkVerProcess.getInputStream()))
+                .lines()
+                .collect(Collectors.joining(" "))
+                .split(" ")[2].substring(1);
+
+        System.out.println("MongoDB version: " + dbVer);
+        checkVerProcess.destroy();
+        return Version.valueOf(dbVer);
+    }
+
+    public List<String> getMongoServerStartCommand(final String mongoExe,
+                                                   final boolean useWiredTiger,
+                                                   final Version dbVer) {
+        List<String> command = new LinkedList<String>();
+        command.addAll(Arrays.asList(mongoExe, "--port", "" + port,
+                "--dbpath", tempDir.resolve(DATA_DIR).toString()));
+
+        // In version 3.6, the --nojournal option is deprecated
+        if (dbVer.lessThanOrEqualTo(MONGO_DB_3)) {
+            command.addAll(Arrays.asList("--nojournal"));
+        }
+        if (useWiredTiger) {
+            command.addAll(Arrays.asList("--storageEngine", "wiredTiger"));
```

Comment:
If you change the matrix so that Mongo7 gets this command will it work?

Reply:
You mean v7 + wired_tiger: 'true'?

Reply:
yeah

Reply:
Wiredtiger is the only choice from about 4.2 or so so I'm wondering if it's silently ignored or throws and error

Reply:
v7 + wired_tiger: 'true' passed. 
 Let me know if you want to keep it in the test.yml

Reply:
Nah, no need for it. I just wanted to be sure that things wouldn't break once we remove support for older mongo versions

Reply:
👍

Reply:
:+1: